capa/features/extractors/dnfile/helpers.py

```diff
@@ -196,6 +221,60 @@ def get_dotnet_managed_methods(pe: dnfile.dnPE) -> Iterator[DnMethod]:
             yield DnMethod(token, row.TypeNamespace, row.TypeName, index.row.Name)
 
 
+def get_dotnet_fields(pe: dnfile.dnPE) -> Iterator[DnProperty]:
+    """get fields from TypeDef table"""
+    for row in iter_dotnet_table(pe, "TypeDef"):
+        for index in row.FieldList:
+            token = calculate_dotnet_token_value(index.table.number, index.row_index)
+            yield DnProperty(token, row.TypeNamespace, row.TypeName, index.row.Name)
+
+
+def get_dotnet_property_map(
+    pe: dnfile.dnPE, property_row: dnfile.mdtable.PropertyRow
+) -> Optional[dnfile.mdtable.TypeDefRow]:
+    """get property map from PropertyMap table
+
+    see https://www.ntcore.com/files/dotnetformat.htm
+
+    21 - PropertyMap Table
+        List of Properties owned by a specific class.
+            Parent (index into the TypeDef table)
+            PropertyList (index into Property table). It marks the first of a contiguous run of Properties owned by Parent. The run continues to the smaller of:
+                the last row of the Property table
+                the next run of Properties, found by inspecting the PropertyList of the next row in this PropertyMap table
+    """
+    for row in iter_dotnet_table(pe, "PropertyMap"):
+        for index in row.PropertyList:
+            if index.row.Name == property_row.Name:
+                return row.Parent.row
+    return None
+
+
+def get_dotnet_properties(pe: dnfile.dnPE) -> Iterator[Tuple[DnProperty, DnPropertyAccessType]]:
+    """get property from MethodSemantics table
+
+    see https://www.ntcore.com/files/dotnetformat.htm
+
+    24 - MethodSemantics Table
+        Links Events and Properties to specific methods. For example one Event can be associated to more methods. A property uses this table to associate get/set methods.
+            Semantics (a 2-byte bitmask of type MethodSemanticsAttributes)
+            Method (index into the MethodDef table)
+            Association (index into the Event or Property table; more precisely, a HasSemantics coded index)
+    """
+    for row in iter_dotnet_table(pe, "MethodSemantics"):
+        typedef_row = get_dotnet_property_map(pe, row.Association.row)
+        if typedef_row is not None:
+            token = calculate_dotnet_token_value(row.Method.table.number, row.Method.row_index)
+            type = (
+                DnPropertyAccessType.SET
+                if row.Semantics.msSetter
+                else DnPropertyAccessType.GET
+                if row.Semantics.msGetter
+                else DnPropertyAccessType.OTHER
```

Comment:
Do we know what the spec says about properties here? Is it possible to have an "other"? Or will it always be "get" or "set"? I'm just curious as to how we imagine "other" may fit into capa or if we even need it?